pycbc/events/stat.py

```diff
@@ -599,12 +603,75 @@ def reassign_rate(self, ifo):
 
+class ExpFitSGFgBgRateStatistic(PhaseTDStatistic, ExpFitSGBgRateStatistic):
+    def __init__(self, files):
+        # read in background fit info and store it, also use newsnr_sgveto
+        ExpFitSGBgRateStatistic.__init__(self, files)
+        # Use PhaseTD statistic single.dtype
+        PhaseTDStatistic.__init__(self, files)
+        self.single_dtype.append(('benchmark_sigma', numpy.float32))
+        self.benchmark_sigma = 3.0 * numpy.log(numpy.amin([self.fits_by_tid[ifo]['median_sigma']
+                                   for ifo in ['H1', 'L1']], axis=0))
+
+        self.get_newsnr = ranking.get_newsnr_sgveto
+
+    def single(self, trigs):
+        # single-ifo stat = log of noise rate
+        sngl_stat = self.lognoiserate(trigs)
+        # populate other fields to calculate phase/time/amp consistency
+        singles = numpy.zeros(len(sngl_stat), dtype=self.single_dtype)
+        singles['snglstat'] = sngl_stat
+        singles['coa_phase'] = trigs['coa_phase'][:]
+        singles['end_time'] = trigs['end_time'][:]
+        singles['sigmasq'] = trigs['sigmasq'][:]
+        singles['snr'] = trigs['snr'][:]
+        try:
+            tnum = trigs.template_num  # exists if accessed via coinc_findtrigs
+            ifo = trigs.ifo
+        except AttributeError:
+            tnum = trigs['template_id']  # exists for SingleDetTriggers
+            # Should only be one ifo fit file provided
+            assert len(self.ifos) == 1
+            ifo = self.ifos[0]
+        singles['benchmark_sigma'] = self.benchmark_sigma[tnum]
+        return numpy.array(singles, ndmin=1)
+
+    def coinc_multiifo(self, s, slide,
+                       step, **kwargs): # pylint:disable=unused-argument
+        sngl_rates = {ifo: sngl_data['snglstat'] for ifo, sngl_data in
+                      s.items()}
+        ln_noise_rate = coinc_rate.combination_noise_lograte(
+                                  sngl_rates, kwargs['time_addition'])
+
+        network_sigma = 1.5 * numpy.log(numpy.amin(
+                            [s[ifo]['sigmasq']
+                             for ifo in s.keys()], axis=0))
+        
+        # logsignalrate function inherited from PhaseTDStatistic
+        # - for now, only use H-L consistency
+        ifos = s.keys()
+        if 'H1' in ifos and 'L1' in ifos:
+            logr_s = self.logsignalrate(s['H1'], s['L1'], slide, step)
+            if len(logr_s):
+                logging.info('Applying HL phase/time/amp consistency')
+            # makeshift factor to compensate HL(V) coincs which are penalized
+            # on average by p/t/a vs HV, LV which are not
+            logr_s = logr_s + 4.5
+        else:
+            logr_s = numpy.zeros_like(s['V1']['snr'])
+            if len(logr_s):
+                logging.info('HV or LV coinc, no phase/time-amp/consistency')
+        loglr = logr_s - ln_noise_rate + self.benchmark_lograte \
+                     + network_sigma - s[ifos[0]]['benchmark_sigma']
```

Comment:
it's not clear to me why this is getting the 'benchmark_sigma' via the singles information - which looks as if it might be ifo-dependent - as far as I can see, benchmark_sigma is a function only of the template not the ifo. 
I guess if there is no way for the coinc() method to directly tell which template it is dealing with this is a reasonable workaround, but will need a bit of commenting.

Reply:
The second one: this was the only way I could see to get the template ID into the `multiifo_coinc()` function. Comment has been added